### PR TITLE
54105 - Appointment List Re-design - Make IDs unique

### PR DESCRIPTION
**Pull request by vbahinwillit**
## Summary
This PR fixes the 508 'duplicate id' issue. See following link for how to test.

https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review#axe

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#54105

## Testing done
Axe DevTools

## Screenshots
Screen Shot 2023-03-08 at 4 42 09 PM
Screen Shot 2023-03-08 at 4 42 46 PM
Screen Shot 2023-03-08 at 4 43 14 PM

## What areas of the site does it impact?
VAOS

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature
- [ ]  [Accessibility foundational testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/vaos/appointment-list/components/AppointmentsPageV2/AppointmentColumnLayout.jsx [hunk 1]
```diff
@@ -128,7 +127,7 @@ export default function AppointmentColumnLayout({
           >
             <va-link
               className="vaos-appts__focus--hide-outline"
-              aria-describedby="vaos-appts__detail"
+              aria-describedby={`aos-appts__detail-${data.id}`}
```
Comment (jenniemc) [inline src/applications/vaos/appointment-list/components/AppointmentsPageV2/AppointmentColumnLayout.jsx:130]:
```suggestion
              aria-describedby={`vaos-appts__detail-${data.id}`}
```

File: src/applications/vaos/appointment-list/components/PastAppointmentsListV2/PastAppointmentsDateDropdown.jsx [hunk 2]
```diff
@@ -49,7 +49,7 @@ export default function PastAppointmentsDateDropdown({
           featureStatusImprovement,
         })}
         id="date-dropdown"
-        value={dateRangeIndex}
+        value={dateRangeIndex.toString()}
```
Comment (jenniemc) [inline src/applications/vaos/appointment-list/components/PastAppointmentsListV2/PastAppointmentsDateDropdown.jsx:52]: Q: why does it need to convert to string?

**Review by jenniemc — DISMISSED** [review]
One change and one question.

File: src/applications/vaos/appointment-list/components/PastAppointmentsListV2/PastAppointmentsDateDropdown.jsx (hunk 2 above)
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/PastAppointmentsListV2/PastAppointmentsDateDropdown.jsx:52]: This fixes props warning message about number provided when string was expected.

**Review by simiadebowale — APPROVED** [review]

**Review by jenniemc — APPROVED** [review]
LGTM!

File: src/applications/vaos/appointment-list/components/AppointmentsPageV2/RequestAppointmentLayout.jsx [hunk 3]
```diff
@@ -36,8 +36,8 @@ export default function RequestAppointmentLayout({ appointment }) {
     <ListItem appointment={appointment} borderTop status="pending">
       <AppointmentFlexGrid idClickable={idClickable} link={link}>
         <AppointmentColumn
-          id="vaos-appts__column--2"
           className={classNames(
+            'vaos-appts__column--2',
```
Comment (vbahinwillit) [inline src/applications/vaos/appointment-list/components/AppointmentsPageV2/RequestAppointmentLayout.jsx:40]: Please remove. Looks like some changes from another PR have been overwritten.

**Pull request merged**